### PR TITLE
Replace % operator and '.format' with Python 3's f-strings

### LINKED ISSUE #1792: Replace % operator and `.format` with Python 3's f-strings
Instead of `'%s is my name' % name` or `'{} is my name'.format(name)`, we should use the much more readable `f'{name} is my name'`.

### PR DESCRIPTION
**Pull request by rv404674**
Fixes #1792.

### CODE REVIEW

**Comment by rv404674** [comment]
hey @marco-c , please can you review it when you get some time. Thanks.

File: src/bugbug/data/bugbug_data/retriever.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def go(self):
 
         # Index the task in the TaskCluster index.
         self.index_service.insertTask(
-            'project.releng.services.project.{}.bugbug_data.latest'.format(secrets[secrets.APP_CHANNEL]),
+            f'project.releng.services.project.{secrets[secrets.APP_CHANNEL]}',
```
Comment (marco-c) [inline src/bugbug/data/bugbug_data/retriever.py]: You removed part of the string here

**Comment by marco-c** [comment]
@rv404674 let me know when you're done!

**Comment by rv404674** [comment]
@marco. Sorry for the delayed update. Yeah sure I will tell you once I am done . It is requiring a lot of time because there are many occurrences of the old format.

**Comment by rv404674** [comment]
hey @marco-c can you please tell me why the build is failing. I am not able to understand the error.
https://tools.taskcluster.net/groups/FDe6Bd4EQWm5rSY9ufcT6g/tasks/FDe6Bd4EQWm5rSY9ufcT6g/runs/0/logs/public%2Flogs%2Flive.log#L485

**Comment by rv404674** [comment]
hey @marco-c . The build is completed (successfull). Can you please take a look when you get some time.

File: lib/backend_common/backend_common/auth.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def __init__(self, token, userinfo):
         self.token = token
         self.userinfo = userinfo
 
-        logger.info('Init user {}'.format(self.get_id()))
+        logger.info(f'Init user {self.get_id()}')
```
Comment (La0) [inline lib/backend_common/backend_common/auth.py]: I don't like seeing a function call in the f-string (from a style point of view), looks sketchy and less readable. What do you think @marco-c  ?

File: lib/backend_common/backend_common/auth.py [hunk 3]
```diff
@@ -337,7 +337,7 @@ def from_relengapi_permission(_permission):
     permission = _permission.strip().replace('.', '/')
     for prefix, project in RELENGAPI_PROJECT_PERMISSION_MAPPING.items():
         if permission.startswith(prefix):
-            return 'project:releng:services/{}{}'.format(project, permission[len(prefix):])
+            return f'project:releng:services/{project}{permission[len(prefix):]}'
```
Comment (La0) [inline lib/backend_common/backend_common/auth.py]: the second part is also hard to read, maybe use a variable

File: lib/backend_common/backend_common/db.py [hunk 4]
```diff
@@ -84,7 +84,7 @@ def init_database(app):
         # Needed to init potential migrations later on
         # Use a separate alembic_version table per app
         options = {
-            'version_table': '{}_alembic_version'.format(app.import_name),
+            f'version_table': '{app.import_name}_alembic_version',
```
Comment (La0) [inline lib/backend_common/backend_common/db.py]: On the other hand, this looks great !

File: lib/please_cli/please_cli/run.py [hunk 5]
```diff
@@ -302,7 +300,7 @@ def cmd(ctx,
         )
 
         for require in project_config.get('requires', []):
-            env_name = '{}_URL'.format(please_cli.utils.normalize_name(require).upper())
+            env_name = f'{please_cli.utils.normalize_name(require).upper()}_URL'
```
Comment (La0) [inline lib/please_cli/please_cli/run.py]: hard to read too

File: src/codecoverage/backend/codecoverage_backend/services/codecov.py [hunk 6]
```diff
@@ -41,22 +41,22 @@ def _get(url=''):
 
     @staticmethod
     async def get_file_coverage(changeset, filename):
-        async with CodecovCoverage._get('/src/{}/{}'.format(await get_github_commit(changeset), filename)) as r:
+        async with CodecovCoverage._get(f'/src/{await get_github_commit(changeset)}/{filename}') as r:
             try:
                 data = await r.json()
             except Exception as e:
                 logger.exception(e)
-                raise CoverageException('Can\'t parse codecov.io report for %s@%s (response: %s)' % (filename, changeset, r.text))
```
Comment (La0) [inline src/codecoverage/backend/codecoverage_backend/services/codecov.py]: from python 2.5 to python 3.7 :+1:

File: lib/backend_common/backend_common/auth.py [hunk 7]
```diff
@@ -186,9 +186,9 @@ def __init__(self, claims, authenticated_email=None, permissions=[], token_data=
     def get_id(self):
         parts = ['token', self.claims['typ']]
         if 'jti' in self.claims:
-            parts.append('id={}'.format(self.claims['jti']))
+            parts.append(f"id={self.claims['jti']}")
```
Comment (marco-c) [inline lib/backend_common/backend_common/auth.py]: @La0 @garbas what is your opinion here? Should we 1) change the string enclosing quotes to " so that we can keep ' for the dict; 2) keep the string enclosing to ' and change the dict to "; 3) don't use f-strings when we have a dict?

Comment (La0) [reply lib/backend_common/backend_common/auth.py]: I vote 2

File: lib/backend_common/backend_common/auth.py (hunk 2 above)
Comment (marco-c) [reply lib/backend_common/backend_common/auth.py]: Yeah, agreed.

File: lib/please_cli/please_cli/run.py (hunk 5 above)
Comment (marco-c) [reply lib/please_cli/please_cli/run.py]: Indeed, in cases like this it'd be better to have a temporary variable, or to keep format

Comment (rv404674) [reply lib/please_cli/please_cli/run.py]: I have kept the old format in cases like these.

**Comment by rv404674** [comment]
hey @marco-c , @La0 i have incorporated the changes suggested by you guys. I am very sorry for taking such a long time. Can you please review it when you get some time.

File: lib/backend_common/backend_common/auth0.py [hunk 8]
```diff
@@ -101,9 +101,7 @@ def auth0_check():
         'Missing code in payload'
 
     # Exchange code for tokens
-    url = 'https://{}/oauth/token'.format(
-        flask.current_app.config.get('AUTH_DOMAIN')
-    )
+    url = f"https://{flask.current_app.config.get('AUTH_DOMAIN')}/oauth/token"
```
Comment (La0) [inline lib/backend_common/backend_common/auth0.py]: Function in the formatted string: please revert this change.

File: lib/cli_common/cli_common/log.py [hunk 9]
```diff
@@ -140,7 +140,7 @@ def init_logger(project_name,
         channel = os.environ.get('APP_CHANNEL')
 
     if channel and channel not in CHANNELS:
-        raise Exception('Initilizing logging with channel `{}`. It should be one of: {}'.format(channel, ', '.join(CHANNELS)))
+        raise Exception('Initializing logging with channel `{}`. It should be one of: {}'.format(channel, ', '.join(CHANNELS)))
```
Comment (La0) [inline lib/cli_common/cli_common/log.py]: :+1:

File: lib/please_cli/please_cli/decision_task.py [hunk 10]
```diff
@@ -405,7 +405,7 @@ def cmd(ctx,
     with click_spinner.spinner():
         task = taskcluster_queue.task(task_id)
         if 'taskGroupId' not in task:
-            please_cli.utils.check_result(1, 'taskGroupId does not exists in task: {}'.format(json.dumps(task)))
+            please_cli.utils.check_result(1, f'taskGroupId does not exists in task: {json.dumps(task)}')
```
Comment (La0) [inline lib/please_cli/please_cli/decision_task.py]: Function in the formatted string: please revert this change.

File: lib/please_cli/please_cli/deploy.py [hunk 11]
```diff
@@ -130,14 +130,14 @@ def cmd_S3(ctx,
             if not os.path.exists(please_cli.config.TMP_DIR):
                 os.makedirs(please_cli.config.TMP_DIR)
             tmp_dir = tempfile.mkdtemp(
-                prefix='copy-of-result-{}-'.format(project.replace('/', '-')),
+                prefix=f"copy-of-result-{project.replace('/', '-')}-",
```
Comment (La0) [inline lib/please_cli/please_cli/deploy.py]: Function in the formatted string: please revert this change.

File: lib/please_cli/please_cli/deploy.py [hunk 12]
```diff
@@ -158,7 +158,7 @@ def cmd_S3(ctx,
                 if csp:
                     index_html = index_html.replace(
                         'font-src \'self\';',
-                        'font-src \'self\'; connect-src {};'.format(' '.join(csp)),
+                        f"font-src \'self\'; connect-src {' '.join(csp)};",
```
Comment (La0) [inline lib/please_cli/please_cli/deploy.py]: Function in the formatted string: please revert this change.

File: lib/please_cli/please_cli/run.py [hunk 13]
```diff
@@ -267,7 +265,7 @@ def cmd(ctx,
 
         # XXX: once we move please_cli.config.PROJECTS to nix we wont need this
         for require in project_config.get('requires', []):
-            env_name = 'WEBPACK_{}_URL'.format(please_cli.utils.normalize_name(require).upper())
+            env_name = f'WEBPACK_{please_cli.utils.normalize_name(require).upper()}_URL'
```
Comment (La0) [inline lib/please_cli/please_cli/run.py]: Function in the formatted string: please revert this change.

File: lib/please_cli/please_cli/shell.py [hunk 14]
```diff
@@ -139,7 +139,7 @@ def cmd(project, zsh, quiet, command, nix_shell,
         envs['TASKCLUSTER_ACCESS_TOKEN'] = taskcluster_access_token
 
     for require in project_config.get('requires', []):
-        env_name = '{}_URL'.format(please_cli.utils.normalize_name(require).upper())
+        env_name = f'{please_cli.utils.normalize_name(require).upper()}_URL'
```
Comment (La0) [inline lib/please_cli/please_cli/shell.py]: Function in the formatted string: please revert this change.

File: src/codecoverage/backend/codecoverage_backend/services/coveralls.py [hunk 15]
```diff
@@ -16,12 +16,12 @@
...
     @staticmethod
     def _get(url=''):
-        return aiohttp.request('GET', 'https://coveralls.io{}'.format(url))
+        return aiohttp.request('GET', f'https://coveralls.io{url}')
 
     @staticmethod
     @alru_cache(maxsize=2048)
     async def get_coverage(changeset):
-        async with CoverallsCoverage._get('/builds/{}.json'.format(await get_github_commit(changeset))) as r:
+        async with CoverallsCoverage._get(f'/builds/{await get_github_commit(changeset)}.json') as r:
```
Comment (La0) [inline src/codecoverage/backend/codecoverage_backend/services/coveralls.py]: Function in the formatted string: please revert this change.

File: src/codecoverage/backend/codecoverage_backend/services/coveralls.py [hunk 16]
```diff
@@ -34,15 +34,15 @@ def _get(url=''):
 
     @staticmethod
     async def get_file_coverage(changeset, filename):
-        async with CoverallsCoverage._get('/builds/{}/source.json?filename={}'.format(await get_github_commit(changeset), filename)) as r:
+        async with CoverallsCoverage._get(f'/builds/{await get_github_commit(changeset)}/source.json?filename={filename}') as r:
```
Comment (La0) [inline src/codecoverage/backend/codecoverage_backend/services/coveralls.py]: Function in the formatted string: please revert this change.

**Comment by rv404674** [comment]
Hey @La0 , i think it is done. Can you please take a look.

**Comment by La0** [comment]
Thanks Rahul, it was not an easy one.

**Comment by rv404674** [comment]
@La0 . No Problem at all. I enjoyed doing it. Although it took a lot of time.
